pyadjoint/checkpointing.py

```diff
@@ -78,9 +84,14 @@ def __init__(self, schedule, tape):
         self.forward_schedule.append(self._current_action)
         # Tell the tape to only checkpoint input data until told otherwise.
         self.tape.latest_checkpoint = 0
-        self.end_timestep(-1)
         self._keep_init_state_in_work = False
-        self._adj_deps_cleaned = False
+        # The user can manually invoke the garbage collector if Python fails to
+        # track and clean all checkpoint objects in memory properly.
```

Comment:
This is confusing because setting `gc_timestep_frequency` suggests that GC is being run automatically, whereas here you say manually